### DISCUSSION TITLE
MakeSet en Kruskal

### DISCUSSION THREAD

**Discussion by sofiacorreabusquets** [IIC2133-PUC/2024-1#196] [Preguntas materia] [answered]
Hola!

Estoy viendo el pseudocódigo de Kruskal en el material de clase, y no entiendo la utilidad de la línea 3. ¿Me lo podrían explicar, por favor?
![image](https://github.com/IIC2133-PUC/2024-1/assets/61999464/4ddc91c1-4f7e-44e9-bad3-46540dbf4614)
Muchas gracias de antemano :)

**Comment by Aleksandsk** [comment, marked as answer]
Para verificar que una arista no forme un ciclo en el árbol, Kruskal utiliza conjuntos disjuntos, donde si los nodos de la arista a verificar están en conjuntos distintos (es decir, ambos tienen un distinto representante), considerar esta arista no forma un ciclo, por lo que si se agrega al árbol y los nodos de la arista pasan a pertenecer al mismo conjunto, en caso contrario la arista forma un ciclo, entonces no se considera.

Entonces para implementar esa lógica, al principio hacemos MakeSet(v) en todos los nodos del grafo original, lo que significa que cada nodo lo estas considerando como un conjunto, donde el mismo nodo es el representante del conjunto, y a medida que va iterando el algoritmo, los conjuntos se van a ir uniendo hasta que todos los nodos queden con el mismo conjunto, es decir, tengan el mismo representante.

Espero que esto aclare tu duda!

**Reply by sofiacorreabusquets** [reply]
Sí, muchas gracias!
